src/QuickMarcEditor/QuickMarcEditor.js

```diff
@@ -76,7 +76,7 @@ const QuickMarcEditor = ({
   const showCallout = useShowCallout();
   const [records, setRecords] = useState([]);
   const [isDeleteModalOpened, setIsDeleteModalOpened] = useState(false);
-  const [authorityLinkDetails, setAuthorityLinkDetails] = useState(null);
+  const [linkedRecords, setLinkedRecords] = useState(null);
```

Comment:
I think we can remove this `useState` and `useEffect` on line 318 altogether
It basically takes all items from `records` that have `_isLinked` property and saves it to `linkedRecords`
We can achieve the same with `useMemo`:
```
const linkedRecords = useMemo(() => records.filter(record => record._isLinked), [records]);
```

Reply:
The intention of having the state is to control open status of modal and also handle confirm/cancel on the modal.
Open or close status of the modal needs to be evaluated only at the initial records load. 
Hence, I think we need the state and useEffect here.

Reply:
Ok, that makes sense. In this case could you change the state so instead of storing the array of linked records it is a boolean values `isUnlinkRecordsModalOpen` or something like this?
I'd like to do it because if `useEffect` only runs once - `records` array and `linkedRecords` can become un-synched if user links some other records and I'd like for `records` to be the source of truth.
With the boolean state there can be no doubt about where the source of truth for records is

Reply:
Yes, I understand your point, but this was intentional. I did give it a thought while working too.
I had the below ideas -
1. state should handle the modal status
2. state should handle confirm/cancel clicks
3. it should never allow the modal to be invoked again - this is reflected from the conditions in useEffect hook on line # 319

Reply:
As discussed, updated the approach. please review